src/pages/ThorChainLP/Pool/components/PoolChart.tsx

```diff
@@ -1,34 +1,130 @@
 import { Button, ButtonGroup, Center, Flex, Stack } from '@chakra-ui/react'
-import { FaChartSimple } from 'react-icons/fa6'
-import { useTranslate } from 'react-polyglot'
-import { RawText } from 'components/Text'
+import { thorchainAssetId } from '@shapeshiftoss/caip'
+import { useQuery } from '@tanstack/react-query'
+import type { SingleValueData } from 'lightweight-charts'
+import { useCallback, useMemo, useState } from 'react'
+import { reactQueries } from 'react-queries'
+import type { Interval } from 'react-queries/queries/midgard'
+import { SimpleChart } from 'components/SimpleChart'
+import { fromThorBaseUnit } from 'lib/utils/thorchain'
+import type {
+  MidgardSwapHistoryResponse,
+  MidgardTvlHistoryResponse,
+} from 'lib/utils/thorchain/lp/types'
+import { selectMarketDataByAssetIdUserCurrency } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+const swapHistoryToChartData = (
+  swapHistory: MidgardSwapHistoryResponse | undefined,
+  runePrice: string,
+): SingleValueData<number>[] => {
+  if (!swapHistory) return []
+
+  return swapHistory.intervals.map(interval => {
+    const intervalVolume2FiatUserCurrency = fromThorBaseUnit(interval.totalVolume)
+      .times(runePrice)
+      .toFixed()
+
+    return {
+      time: Number(interval.startTime),
+      value: Number(intervalVolume2FiatUserCurrency),
+    }
+  })
+}
+
+const tvlToChartData = (
+  tvl: MidgardTvlHistoryResponse,
+  runePrice: string,
+  thorchainNotationAssetId: string,
+): SingleValueData<number>[] =>
+  tvl.intervals.map(interval => {
+    const poolDepth = interval.poolsDepth.find(pool => pool.pool === thorchainNotationAssetId)
+    const poolTotalDepth = poolDepth?.totalDepth ?? '0'
+
+    const tvlFiat = fromThorBaseUnit(poolTotalDepth).times(runePrice).toFixed()
+    return {
+      time: Number(interval.startTime),
+      value: Number(tvlFiat),
+    }
+  })
+
+// @ts-ignore we can't make this a partial record as we need to use this as a tuple to spread as useQuery params
+const INTERVAL_PARAMS_BY_INTERVAL: Record<Interval | 'all', [Interval, number]> = {
+  day: ['hour', 24],
+  week: ['hour', 24 * 7],
+  month: ['day', 7 * 4],
+  all: ['month', 24],
```

Comment:
suggestion we change this to 1 year since it's not really `all` after 24 months of history is available

Reply:
Could absolutely do, it's more about the mainnet not being up for so many years just yet - could go with either one year (@reallybeard q) or with an arbitrary long timeframe if the API doesn't rug us

Reply:
I'm cool with changing to 1Y, can always update the label if we ever get more than one year of data later.

Reply:
https://github.com/shapeshift/web/pull/6649 @reallybeard whichever solution we go with will be implemented in that PR 🙏🏽